### ISSUE TITLE
depot rendering

### ISSUE THREAD

**Issue by hyn0801** [tomsilver/pddlgym#25]
@tomsilver 
Thank you for sharing !
I wonder is there a file for 'depot' render function? 
or give some advice how to Implement it.
Thanks.

**Comment by tomsilver** [comment]
Hi @hyn0801 , thanks for your interest! We do not have a depot render function right now; it would be great if you wanted to contribute one.

Adding a new render function can be done as follows:
1. Add a new file in `rendering/` with a function `render` that takes in a set of literals and returns an image (e.g. a numpy array). See [here](https://github.com/tomsilver/pddlgym/blob/master/rendering/blocks.py#L112) for an example.
2. Import your render function [here](https://github.com/tomsilver/pddlgym/blob/master/rendering/__init__.py) `as depot_render`
3. Add `'render': depot_render` to the kwargs for `depot` [here](https://github.com/tomsilver/pddlgym/blob/master/__init__.py)

For implementing the render function, there are some utilities in `rendering` that may be useful. For example, if you want to organize the image as a grid of sprites, `render_from_layout` [here](https://github.com/tomsilver/pddlgym/blob/master/rendering/utils.py#L50) will do a lot of the work for you. See [here](https://github.com/tomsilver/pddlgym/blob/master/rendering/sokoban.py) for one example of that function in use.

Let us know if you have any questions!

**Comment by hyn0801** [comment]
Thanks you.
I will try for it.
I have found the depot.pddl file is defined only 1 types named "thing", different from IPC2002 here[http://ipc02.icaps-conference.org/](url), did you write these pddl domain files by yourself, is there any rules to add myowe pddl domain file?

**Comment by tomsilver** [comment]
@hyn0801 ah yes, thanks for asking. back when we added the depot.pddl file, pddlgym did not support hierarchical types, so we made some changes. but since then, we have added support for hierarchical types, so the original depot.pddl should work fine.

we went ahead and added the original depot.pddl file and problems from IPC2002:
https://github.com/tomsilver/pddlgym/commit/a71ab6424997dede6625fcb5c3f1b3c9747cbcf4

the README has information about adding your own PDDL domain file, if you want to add others. let us know if you run into any issues!
